homeassistant/components/philips_js/light.py

```diff
@@ -217,16 +216,16 @@ def _calculate_effect(self) -> AmbilightEffect:
         return AmbilightEffect(EFFECT_MODE, self._tv.ambilight_mode, None)
 
     @property
-    def color_mode(self):
+    def color_mode(self) -> ColorMode:
         """Return the current color mode."""
         current = self._tv.ambilight_current_configuration
         if current and current["isExpert"]:
-            return COLOR_MODE_HS
+            return ColorMode.HS
 
         if self._tv.ambilight_mode in ["manual", "expert"]:
-            return COLOR_MODE_HS
+            return ColorMode.HS
 
-        return COLOR_MODE_ONOFF
+        return ColorMode.ONOFF
```

Comment:
This is still not right since it shouldn't set this unless it never supports HS

I don't know enough about the integration to suggest a fix though

Reply:
I have reinstated ONOFF, to match the original functionnality.
Let's see if @elupus has another idea.

Reply:
Not sure I follow. That is the current color mode. It depends on the effect state. Some effect states can't set any colors at all.

Reply:
The problem is that self._attr_supported_color_modes should be made dynamic. docs say not to add ONOFF if HS is added

https://developers.home-assistant.io/docs/core/entity/light?_highlight=color#color-modes

Reply:
Hmm. Okey. I suppose it could just report None as color yet still say HS mode and just ignore any color being requested.

Reply:
If you don't know what color mode its in, there is `ColorMode.UNKNOWN`

Reply:
I know the color mode. It is OnOff.. Or it only support turning on or off when effects are set as ambient lighting.

Reply:
This is how I tweaked it for Tuya: 4f3fd6308ce67d6cd6f17f92c85cac41c431c5c4

Reply:
This onoff should hopefully be replaced with: https://github.com/home-assistant/core/pull/55450 if something like that is accepted. This is a dynamic mode that changes based on settings.